lib/matplotlib/axes/_axes.py

```diff
@@ -5049,11 +5093,29 @@ def reduce_C_function(C: array) -> float
                 ymin = 10.0 ** ymin
                 ymax = 10.0 ** ymax
                 self.set_yscale(yscale)
+
+            if xscale == 'symlog':
+                polygons[:, :, 0] = inv_symlog_trans['x'].transform_non_affine(
+                    polygons[:, :, 0]
+                )
+                xmin, xmax = inv_symlog_trans['x'].transform_non_affine(
+                    np.array([xmin, xmax])
+                )
+                self.set_xscale(xscale, **xscale_kwargs)
```

Comment:
Also, the defaults from earlier would not be passed here?

Reply:
you're right.